### PR TITLE
[CROSSDATA-114] [Mongo Datasource] Ensure subdoc fields with correct order

### PR DESCRIPTION
**Pull request by pmadrigal**

### CODE REVIEW

File: spark-mongodb/src/main/scala/com/stratio/datasource/mongodb/schema/MongodbRowConverter.scala [hunk 1]
```diff
@@ -80,12 +81,22 @@ with Serializable {
...
   def rowAsDBObject(row: Row, schema: StructType): DBObject = {
-    val attMap: Map[String, Any] = schema.fields.zipWithIndex.map {
+    val attMap = toListMap(schema.fields.zipWithIndex.map {
       case (att, idx) => (att.name, toDBObject(row(idx),att.dataType))
-    }.toMap
+    })
     attMap
   }
 
+  def toListMap( value: Array[(String, Any)]): ListMap[String, Any] = {
```
Comment (pfcoperez) [inline spark-mongodb/src/main/scala/com/stratio/datasource/mongodb/schema/MongodbRowConverter.scala]: I would try `ListMap(arr:_*)` where `arr` is a `Seq`

**Comment by darroyocazorla** [comment]
+1

**Comment by pfcoperez** [comment]
:+1:

**Comment by ccaballe** [comment]
+1 Awesome!

**Pull request merged**